File: java/depthFirstSearch.java
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;
 
public class depthFirstSearch{ 
 
	static class Node{
		int data;
		boolean visited;
		List<Node> neighbours;
 
		Node(int data){
			this.data = data;
			this.neighbours=new ArrayList<>();
 
		}
		public void addneighbours(Node neighbourNode){
			this.neighbours.add(neighbourNode);
		}
		public List<Node> getNeighbours() {
			return neighbours;
		}
		public void setNeighbours(List<Node> neighbours) {
			this.neighbours = neighbours;
		}
	}
 
	// Recursive DFS
	public  void dfs(Node node){
		List<Node> neighbours=node.getNeighbours();
        node.visited=true;
		for (int i = 0; i < neighbours.size(); i++) {
			Node n=neighbours.get(i);
			if(n!=null && !n.visited){
				dfs(n);
			}
		}
	}
 
	public static void main(String arg[]){
		int n = 1000;
		Node[] nodes = new Node[n];

		for (int i = 0; i < n; i++){
			nodes[i] = new Node(i);
		}

		for (int i = 0; i < n; i++){
			for (int j = i; j < n; j++){
				nodes[i].addneighbours(nodes[j]);
			}
		}

		depthFirstSearch dfs = new depthFirstSearch();
		final double startTime = System.nanoTime() / 1000000.0;
		dfs.dfs(nodes[7]);
        final double endTime = System.nanoTime() / 1000000.0;

        System.out.println(endTime - startTime);
	}
}
 

File: kt/depthFirstSearch.kt
class Node(var dat: Int) {
    var visited: Boolean = false
    var neighbours = mutableListOf<Node>()

    fun addneighbours(neighbourNode: Node) {
        this.neighbours.add(neighbourNode)
    }

    fun getneighbours() = this.neighbours

    fun setneighbours(neighbours: MutableList<Node>) {
        this.neighbours = neighbours
    }
}

fun dfs(node: Node) {
    val neighbours = node.getneighbours()
    node.visited = true
    for (el in neighbours) {
        if (!el.visited) {
            dfs(el)
        }
    }
}


fun main(){
    val n = 1000
    val nodes = Array(n) { i -> Node(i) }
    for (i in 0 until n) {
        for (j in i until n) {
            nodes[i].addneighbours(nodes[j])
        }
    }
    val time = System.nanoTime()
    dfs(nodes[7])
    val timeMilli = (System.nanoTime() - time) / 1000000.0f

    println("$timeMilli")
}
